crates/router/src/core/refunds_v2.rs

```diff
@@ -0,0 +1,798 @@
+    )
+    .change_context(errors::ApiErrorResponse::RefundAmountExceedsPaymentAmount)?;
+
+    refunds_validator::validate_maximum_refund_against_payment_attempt(
+        &all_refunds,
+        state.conf.refund.max_attempts,
+    )
+    .change_context(errors::ApiErrorResponse::MaximumRefundCount)?;
+
+    let connector = payment_attempt
+        .connector
+        .clone()
+        .ok_or(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("No connector populated in payment attempt")?;
+    let (connector_transaction_id, processor_transaction_data) =
+        ConnectorTransactionId::form_id_and_data(connector_payment_id);
+    let refund_create_req = storage::RefundNew {
+        id: global_refund_id,
+        merchant_reference_id: merchant_reference_id.clone(),
+        external_reference_id: Some(merchant_reference_id.get_string_repr().to_string()),
+        payment_id: req.payment_id,
+        merchant_id: merchant_account.get_id().clone(),
+        connector_transaction_id,
+        connector,
+        refund_type: enums::RefundType::foreign_from(req.refund_type.unwrap_or_default()),
+        total_amount: payment_attempt.get_total_amount(),
+        refund_amount,
+        currency,
+        created_at: common_utils::date_time::now(),
+        modified_at: common_utils::date_time::now(),
+        refund_status: enums::RefundStatus::Pending,
+        metadata: req.metadata,
+        description: req.reason.clone(),
+        attempt_id: payment_attempt.id.clone(),
+        refund_reason: req.reason,
+        profile_id: Some(payment_intent.profile_id.clone()),
+        connector_id: payment_attempt.merchant_connector_id.clone(),
+        charges: None,
+        split_refunds: None,
+        connector_refund_id: None,
+        sent_to_gateway: Default::default(),
+        refund_arn: None,
+        updated_by: Default::default(),
+        organization_id: merchant_account.organization_id.clone(),
+        processor_transaction_data,
+        processor_refund_data: None,
+    };
+
+    let refund = match db
+        .insert_refund(refund_create_req, merchant_account.storage_scheme)
+        .await
+    {
+        Ok(refund) => {
+            Box::pin(schedule_refund_execution(
+                state,
+                refund.clone(),
+                refund_type,
+                merchant_account,
+                key_store,
+                payment_attempt,
+                payment_intent,
+            ))
+            .await?
+        }
+        Err(err) => {
+            if err.current_context().is_db_unique_violation() {
+                Err(errors::ApiErrorResponse::DuplicateRefundRequest)?
+            } else {
+                return Err(err)
+                    .change_context(errors::ApiErrorResponse::RefundNotFound)
+                    .attach_printable("Inserting Refund failed");
+            }
+        }
+    };
+    let unified_translated_message = if let (Some(unified_code), Some(unified_message)) =
```

Comment:
Avoid branching, have a separate function or writing it in idiomatic way. Don't use if let syntax instead pattern using match itself.